st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -286,6 +291,84 @@ def get(self, id, attribute, requester_user):
         return result
 
 
+class ActionExecutionOutputController(ActionExecutionsControllerMixin, ResourceController):
+    supported_filters = {
+        'output_type': 'output_type'
+    }
+    exclude_fields = []
+
+    def get_one(self, id, output_type=None, requester_user=None):
+        # Special case for id == "last"
+        if id == 'last':
+            execution_db = ActionExecution.query().order_by('-id').limit(1).first()
+        else:
+            execution_db = self._get_one_by_id(id=id, requester_user=requester_user,
+                                               permission_type=PermissionType.EXECUTION_VIEW)
+
+        execution_id = str(execution_db.id)
+
+        query_filters = {}
+        if output_type:
+            query_filters['output_type'] = output_type
+
+        def existing_output_iter():
+            # Consume and return all of the existing lines
+            # pylint: disable=no-member
+            output_dbs = ActionExecutionOutput.query(execution_id=execution_id, **query_filters)
+
+            # Note: We return all at once instead of yield line by line to avoid multiple socket
+            # writes and to achieve better performance
+            output = ''.join([output_db.data for output_db in output_dbs])
+            yield six.binary_type(output.encode('utf-8'))
+
+        def new_output_iter():
+            def noop_gen():
+                yield ''
+
+            # Bail out if execution has already completed
+            # TODO: Should we also bail out in PAUSE? Perhaps not a bad idea to avoid potentially
+            # very long opened connection
+            if execution_db.status in action_constants.LIVEACTION_COMPLETED_STATES:
```

Comment:
Yeah, definitely should bail out in PAUSE since the execution is not active.

Reply:
k, will make that change.

Reply:
ac6c00a129e1dc29f8a84a4fab33b94ff3382cd5